overlord/devicestate/devicemgr.go

```diff
@@ -1877,74 +1877,81 @@ func (m *DeviceManager) Systems() ([]*System, error) {
 // including the model assertion, gadget details and encryption info
 // for the given system label.
 func (m *DeviceManager) SystemAndGadgetAndEncryptionInfo(wantedSystemLabel string) (*System, *gadget.Info, *EncryptionSupportInfo, error) {
-	if m.isClassicBoot {
-		return nil, nil, nil, fmt.Errorf("cannot get model and gadget information on a classic boot system")
-	}
+	sys, gadgetInfo, _, _, encInfo, err := m.LoadLabelInformation(wantedSystemLabel)
+	return sys, gadgetInfo, encInfo, err
+}
+
+// LoadLabelInformation loads information for the given label, which
+// includes system, gadget information, gadget and kernel snaps info,
+// gadget and kernel seed snap info, and encryption support info.
+func (m *DeviceManager) LoadLabelInformation(wantedSystemLabel string) (
```

Comment:
my impression looking at this PR and the draft one is that this should become:

```
func (m *DeviceManager) loadSystemAndEssentialSnaps(wantedSystemLabel string, types []snap.Type) (*System, map[snap.Type]*snap.Info, map[snap.Type]*seed.Snap, error)
```

and the rest stay in the higher level bits. It's also not clear it needs to be exported?